### PR TITLE
scaletesting: add bitbucket client for codehost copy

### PR DESCRIPTION
**Pull request by burmudar**
See title
## Test plan
Test locally

### CODE REVIEW

File: dev/scaletesting/codehostcopy/bitbucket.go [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+	"net/url"
+	"strings"
+
+	"github.com/sourcegraph/log"
+	"github.com/sourcegraph/sourcegraph/dev/scaletesting/codehostcopy/bitbucket"
+	"github.com/sourcegraph/sourcegraph/lib/errors"
+)
+
+type BitbucketCodeHost struct {
+	log log.Logger
```
Comment (jhchabran) [inline dev/scaletesting/codehostcopy/bitbucket.go:15]:
```suggestion
	logger log.Logger
```

File: dev/scaletesting/codehostcopy/bitbucket.go [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+	def *CodeHostDefinition
+	c   *bitbucket.Client
+}
+
+func NewBitbucketCodeHost(ctx context.Context, log log.Logger, def *CodeHostDefinition) *BitbucketCodeHost {
+	u, _ := url.Parse(def.URL)
```
Comment (jhchabran) [inline dev/scaletesting/codehostcopy/bitbucket.go:21]: If we're not dealing with the error, we're better off with a string there.

File: dev/scaletesting/codehostcopy/bitbucket.go [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+	c := bitbucket.NewClient(def.Username, def.Password, u)
```
Comment (jhchabran) [inline dev/scaletesting/codehostcopy/bitbucket.go:22]: There's no token based auth? If yes, that means those should be mandatory indeed. I think you talked about that earlier.

File: dev/scaletesting/codehostcopy/bitbucket.go [hunk 4]
```diff
@@ -0,0 +1,102 @@
...
+	}
+}
+
+func getCloneUrl(repo *bitbucket.Repo) (*url.URL, error) {
+	cloneLinks, ok := repo.Links["clone"]
+	if !ok {
+		return nil, errors.Newf("no clone links on repo %s", repo.Name)
+	}
+	for _, l := range cloneLinks {
+		if l.Name == "https" || l.Name == "http" {
```
Comment (jhchabran) [inline dev/scaletesting/codehostcopy/bitbucket.go:37]: Is there a case where http is actually acceptable?

File: dev/scaletesting/codehostcopy/bitbucket/client.go [hunk 5]
```diff
@@ -0,0 +1,312 @@
...
+	}
+
+	return &pageResp, nil
+}
+
+func (c *Client) Get(ctx context.Context, url string) ([]byte, error) {
+	req, _ := http.NewRequestWithContext(ctx, http.MethodGet, url, nil)
+	req.SetBasicAuth(c.username, c.password)
+
+	resp, err := http.DefaultClient.Do(req)
```
Comment (jhchabran) [inline dev/scaletesting/codehostcopy/bitbucket/client.go:103]: Let's adjust the timeouts there, the default one has very loose defaults.

**Review by jhchabran — CHANGES_REQUESTED** [review]

File: dev/scaletesting/codehostcopy/bitbucket/client.go (hunk 5 above)
Comment (burmudar) [reply dev/scaletesting/codehostcopy/bitbucket/client.go:135]: Adjusted to be 15 secs

File: dev/scaletesting/codehostcopy/main.go [hunk 6]
```diff
@@ -12,6 +12,7 @@ import (
 	"github.com/urfave/cli/v2"
 
 	"github.com/sourcegraph/log"
+
```
Comment (burmudar) [inline dev/scaletesting/codehostcopy/main.go:15]:
```suggestion
```

**Review by jhchabran — APPROVED** [review]

**Pull request merged**